src/service-worker.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-undef, max-len, no-restricted-globals, no-underscore-dangle */
+const cachePrefix = 'quiches-wars';
+const cacheStaticName = `${cachePrefix}-static`;
+const cacheRequestName = `${cachePrefix}-requests`;
+const cacheNamesToDeleteOnUpdate = [cacheStaticName, cacheRequestName];
+
+// define a prefix for your cache names. It is recommended to use your project name
+workbox.core.setCacheNameDetails({ prefix: cachePrefix });
+
+// __precacheManifest is the list of resources you want to precache. This list will be generated and imported automatically by workbox during build time
+self.__precacheManifest = [].concat(self.__precacheManifest || []);
+workbox.precaching.precacheAndRoute(self.__precacheManifest, {});
+
+// ADD MORE REQUEST TO CACHE WITH EXAMPLE BELOW
+// workbox.routing.registerRoute(
+//   /.+\\.js$/,
+//   new workbox.strategies.StaleWhileRevalidate(),
+// );
+
+async function installHandler() {
+  const staticFiles = [
+    // Static file route
+  ];
+
+  const requests = [
+    '', // For a GET request
+    { // For specific request
+      url: '', // url to fetch
+      params: {
+        // fetch params
+      },
+    },
+  ];
+
+  const [staticCache, requestCache] = await Promise.all([caches.open(cacheStaticName), caches.open(cacheRequestName)]);
+
+  const hasRequestCache = await caches.has(cacheRequestName);
+
+  if (!hasRequestCache) {
+    throw new Error('Impossible to use ', cacheRequestName, ' cache !');
+  }
+
+  // Add static files to its cache
+  const cachePromises = [staticCache.addAll(staticFiles)];
+
+  // Add request to its cache
+  const reqFetches = [];
+  const reqResult = [];
+
+  for (let reqIndex = 0; reqIndex < requests.length; reqIndex += 1) {
+    const req = requests[reqIndex];
+
+    let reqUrl = req;
+    let reqParams = {};
+
+    if (typeof req !== 'string') {
+      reqUrl = req.url;
+      reqParams = req.params;
+    }
+
+    reqFetches.push(
+      fetch(reqUrl, reqParams)
```

Comment:
Why fetch and not use axios

Reply:
Because it's used in documentation 😉 

Without joke, fetch is more low level so I think it's better for Service Worker who's need high performance.